pkg/ingester/transfer.go

```diff
@@ -281,10 +293,12 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 			return errors.Wrap(err, "TransferTSDB: ClaimTokensFor")
 		}
 
+		receivedBytes.Add(float64(bytesXfer))
 		receivedFiles.Add(float64(filesXfer))
-		level.Error(util.Logger).Log("msg", "Total files xfer", "from_ingester", fromIngesterID, "num", filesXfer)
+		level.Info(util.Logger).Log("msg", "Total xfer", "from_ingester", fromIngesterID, "files", filesXfer, "bytes", bytesXfer)
 
-		return nil
+		// Move the tmpdir to the final location
+		return os.Rename(tmpDir, i.cfg.TSDBConfig.Dir)
```

Comment:
For the sake of correctness, I would suggest you to add an assertion to `TestV2IngesterTransfer`, checking:
1. `i.cfg.TSDBConfig.Dir` is empty (or not existent) before transfer
2. `i.cfg.TSDBConfig.Dir` is not empty and contains the expected userIDs after transfer

If I'm not missing anything, right now that test doesn't guarantee that the transfer occurred to the right location.

Reply:
Added assertion the data is where expected.